### PR TITLE
Country's Statistics App

### PR DESCRIPTION
**Pull request by anita00001**
# In this branch, we have:

## Create a new React App (#1 )
- Created a new react app using `npx create-react-app country-statistic`
- Deleted all boilerplates (including files, images, and texts)

## Setup Redux Store (#2 )
- Setup the redux store `./src/redux/store.js`
- Provided Redux store to React

## Add dependencies for linter check (#3 )
- Added [.github/workflows/linters.yml](https://github.com/microverseinc/linters-config/blob/master/react-redux/.github/workflows/linters.yml)
- Installed ESLint
     - `npm install --save-dev eslint@7.x eslint-config-airbnb@18.x eslint-plugin-import@2.x eslint-plugin-jsx-a11y@6.x eslint-plugin-react@7.x eslint-plugin-react-hooks@4.x @babel/eslint-parser@7.x @babel/core@7.x  @babel/plugin-syntax-jsx@7.x  @babel/preset-react@7.x @babel/preset-react@7.x`
- Added [.eslintrc.json](https://github.com/microverseinc/linters-config/blob/master/react-redux/.eslintrc.json) and [.babelrc](https://github.com/microverseinc/linters-config/blob/master/react-redux/.babelrc) to the root directory of the project
- Installed Stylelint
    - `npm install --save-dev stylelint@13.x stylelint-scss@3.x stylelint-config-standard@21.x stylelint-csstree-validator@1.x`
- Added [.stylelintrc.json](https://github.com/microverseinc/linters-config/blob/master/react-redux/.stylelintrc.json) to the root directory of the project

## Fetch from API (#4 )
- Added Slice for country
- Fetched Country from [rest countries](https://restcountries.com/v3.1/all) API
- Displayed the Country name and country's area on the Home page

## Country Information (#5 )
- Added click actions
- Displayed country information on another page upon clicking on the country name on the main page

## Features to personalize the app (#6 )
- Added a dropdown with continents as options
- Added filter functionality that filters the countries by continent
- Added a search bar that allows one to search a country by its name
- Added five recommendations of countries on the info page

## Icons and Styles (#7 )
- Added react-icons [microphone, setting, return]
- Added flags for respective countries
- Added styles using CSS

## Testing (#8 )
- Added test cases for `App.jsx` --> `App.test.js`
- Added test cases for `NavBar.jsx` --> `NavBar.test.js`
- Added test cases for `Countries.jsx` --> `Countries.test.js`
- Added test cases for `CountryInfo.jsx` --> `CountryInfo.test.js`

## Documentation and Deployment (#9 )
- Deployed the app on Vercel

- Added MIT license
- Updated README

# Other features
- ### No Linters
- ### Use of correct Gitflow
- ### Professional documentation
- ### Best use of JavaScript and CSS

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -240,52 +175,31 @@ Example:
...
-> Write a message to encourage readers to support your project
-
-If you like this project...
+If you like this project give it a rating.
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
 
 <!-- ACKNOWLEDGEMENTS -->
 
 ## 🙏 Acknowledgments <a name="acknowledgements"></a>
```
Comment (ridwanediallo) [inline README.md:192]: - [x] Great job adding some acknowledgement to the appropriate section 👏! However, I suggest giving credit to the original designer, [Nelson Sakwa](https://www.behance.net/sakwadesignstudio), who made the design available for free. This not only meets the project requirements but also shows appreciation for their work. They deserve recognition for making their work accessible to us. :ok_hand:  😊

File: src/components/Countries.jsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                  {item.area}
+                  {' '}
+                  km²
+                </p>
+              </Link>
+            </div>
+          ))}
+        </div>
+      </section>
+    </div>
```
Comment (ridwanediallo) [inline src/components/Countries.jsx:73]: - [x] You did an excellent job of styling the home page 👏 🥳. However, in accordance with the project specifications, I would kindly suggest that you stick to the original design by using the same colours used in the design for the background colours. You have the option of using either the pink or blue colour. It is entirely up to you which one you prefer. :ok_hand: 🤓

#### Project requirement :point_down: 
Screenshot from 2023-08-03 13-38-29


#### Your design :point_down: 
Screenshot from 2023-08-03 13-31-12


#### The original design
Screenshot from 2023-08-03 13-32-55 |  
Screenshot from 2023-08-03 13-45-08

**Review by ridwanediallo — CHANGES_REQUESTED** [review]
# Changes Required ♻️
## Hi @anita00001 , 
AlmostThereReachGIF

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- GitHub flow is used correctly. ✔
- Kudos adding the time to your app. ✔ 💪
- Your app is correctly tested. ✔👏 👏 👏 💪


## Required Changes ♻️

- [x] Please make sure to record a 5-minute video to present your project. Not only is it a project requirement, but it also helps to promote your work.

#### Presentational video 
Screenshot from 2023-08-01 15-37-13




## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 3]
```diff
@@ -45,192 +14,158 @@ For more information on the importance of a professional README for your reposit
...
 - [📝 License](#license)
 
 <!-- PROJECT DESCRIPTION -->
 
-# 📖 [your_project_name] <a name="about-project"></a>
+# 📖 Statistics 0f Countries <a name="about-project"></a>
 
-> Describe your project in 1 or 2 sentences.
+> This is a React app that fetches the country's data from an external API and displays the country name when the app loads. When a user clicks on the country card, it displays the statistics of the selected country. It also allows searching for a country by its name and sorting countries by their continent.
+> [Click here](https://drive.google.com/file/d/1QKiu0nmpFZeMdhakS2x0mmiwvHqFJTwK/view?usp=drive_link) for more details.
```
Comment (anita00001) [inline README.md:36]: The video presentation is linked here. The video is of 4:30 minutes.

**Review by emmiiorji — APPROVED** [review]
Hi @anita00001, 

# Status: Approved ✅

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

<div align='center'>

</div>



## Good Highlights✔️

- Requested changes have been implemented

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**